Extensions/Leaderboards/tests/leaderboardstools.spec.js

```diff
@@ -0,0 +1,55 @@
+describe('Leaderboards', () => {
+  describe('formatPlayerName', () => {
+    it('it returns name if correct', () => {
+      expect(gdjs.evtTools.leaderboards.formatPlayerName('PlayerName')).to.be(
+        'PlayerName'
+      );
+    });
+
+    it('it returns name with underscores instead of whitespaces', () => {
+      expect(gdjs.evtTools.leaderboards.formatPlayerName('Player Name')).to.be(
+        'Player_Name'
+      );
+    });
+
+    it("it doesn't change a name with vertical bars and hyphens", () => {
+      expect(gdjs.evtTools.leaderboards.formatPlayerName('Pla-yer|Name')).to.be(
+        'Pla-yer|Name'
+      );
+    });
+
+    it('it truncates name if longer than 30', () => {
+      expect(
+        gdjs.evtTools.leaderboards.formatPlayerName(
+          'aPlayerNameTh4tIsT00LongToBeSaved'
+        )
+      ).to.be('aPlayerNameTh4tIsT00LongToBeSa');
+    });
+
+    it('it generates a predefined player name with a random number if input is void/wrong type/empty', () => {
+      expect(gdjs.evtTools.leaderboards.formatPlayerName(null))
+        .to.be.a('string')
+        .and.match(/^Player\d{5}/);
+      expect(gdjs.evtTools.leaderboards.formatPlayerName(5))
+        .to.be.a('string')
+        .and.match(/^Player\d{5}/);
+      expect(gdjs.evtTools.leaderboards.formatPlayerName(undefined))
+        .to.be.a('string')
+        .and.match(/^Player\d{5}/);
+      expect(gdjs.evtTools.leaderboards.formatPlayerName(() => {}))
```

Comment:
Typing + the extension parameter types should prevent anything but a string to reach the function.

Reply:
If a user uses an expression, or a JS event, it could store in a variable all sorts of things, couldn't it? Or are there securities that I'm not aware of?

Reply:
A GDevelop expression are "type safe" and will always guarantee that you pass a proper string (otherwise the expression is not valid and won't even be executed, it will return an empty string or 0 to the action/condition using it).

But you're right, a JS event could call this with an improper type, so it's ok to protect against it.